### PR TITLE
changes required by the removal of TeamMembers from SE

### PR DESCRIPTION
**Pull request by makingthematrix**
requires https://github.com/wireapp/wire-android-sync-engine/pull/175
#### APK
[Download build #9124](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9124/artifact/build/artifact/wire-dev-PR969-9124.apk)
[Download build #9126](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9126/artifact/build/artifact/wire-dev-PR969-9126.apk)
[Download build #9127](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9127/artifact/build/artifact/wire-dev-PR969-9127.apk)

### CODE REVIEW

**Comment by makingthematrix** [comment]
build again

**Comment by deanrobertcook** [comment]
build again

**Comment by deanrobertcook** [comment]
build again

File: app/src/main/scala/com/waz/zclient/common/views/ChatheadView.scala [hunk 1]
```diff
@@ -278,7 +279,14 @@ protected class ChatheadController(val setSelectable:            Boolean
...
   }
 
-  val teamMember = Signal.const(false)//TODO: check if it's a team member
+  val teamMember = for {
+    zms <- zMessaging
+    userTeamId <- chatheadInfo.map {
+      case Some(Left(user)) => user.teamId
+      case _ => None
+    }
+    selfTeamId <- zms.teams.selfTeam.map(_.map(_.id))
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/common/views/ChatheadView.scala]: we can just do `zms.teamId.isDefined`

File: app/src/main/scala/com/waz/zclient/fragments/PickUserFragment.scala [hunk 2]
```diff
@@ -322,6 +320,16 @@ class PickUserFragment extends BaseFragment[PickUserFragment.Container]
       }
     })
 
+    (for {
+      z <- zms
+      team <- z.teams.selfTeam
+      accountData <- z.account.accountData
+    } yield (team, accountData)){
+      case (team, accountData) =>
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/fragments/PickUserFragment.scala]: the yield is unnecessary, `Signal.foreach` (a non-yield for-comprehension) just adds a subscriber to signals

File: app/src/main/scala/com/waz/zclient/views/ChatheadWithTextFooter.scala [hunk 3]
```diff
@@ -55,7 +54,8 @@ class ChatheadWithTextFooter(val context: Context, val attrs: AttributeSet, val
     z <- inject[Signal[ZMessaging]]
     uId <- userId
     data <- z.usersStorage.signal(uId)
-  } yield (data, false) //TODO: this false means guest status
+    guests <- z.teams.guests
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/views/ChatheadWithTextFooter.scala]: here we don't need to trigger the instantiation of `Team.guests` - we can just check to see if `z.teamId.isDefined && z.teamId != data.teamId` (guests will have a different teamId)

File: app/src/main/scala/com/waz/zclient/views/ParticipantDetailsTab.scala [hunk 4]
```diff
@@ -47,7 +48,12 @@ class ParticipantDetailsTab(val context: Context, val attrs: AttributeSet, val d
   setOrientation(LinearLayout.VERTICAL)
 
   private val userId = Signal[UserId]()
-  private val isGuest = Signal(false)
+  private val isGuest = for{
+    z <- inject[Signal[ZMessaging]]
+    uId <- userId
+    guests <- z.teams.guests
+  } yield guests.contains(uId)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/views/ParticipantDetailsTab.scala]: here as well

Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/views/ParticipantDetailsTab.scala]: As a matter of fact - let's check to see if the `guests` list is even used, and potentially remove it

**Comment by makingthematrix** [comment]
build again

File: app/src/main/scala/com/waz/zclient/common/views/ChatheadView.scala [hunk 5]
```diff
@@ -278,7 +278,13 @@ protected class ChatheadController(val setSelectable:            Boolean
     case _ => UNCONNECTED
   }
 
-  val teamMember = Signal.const(false)//TODO: check if it's a team member
+  val teamMember = for {
+    zms <- zMessaging
+    userTeamId <- chatheadInfo.map {
+      case Some(Left(user)) => user.teamId
+      case _ => None
+    }
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/common/views/ChatheadView.scala]:
```
isTeam <- userTeamId.map(zms.teams.isGuest).getOrElse(Future.successful(false))
yield isTeam
```

**Pull request merged**